dbms/src/Interpreters/Join.cpp

```diff
@@ -1911,10 +1980,15 @@ void Join::checkTypesOfKeys(const Block & block_left, const Block & block_right)
     }
 }
 
-void Join::joinBlock(Block & block) const
+Block Join::joinBlock(ProbeProcessInfo & probe_process_info) const
 {
-    //    std::cerr << "joinBlock: " << block.dumpStructure() << "\n";
+    Block block = probe_process_info.block;
```

Comment:
```suggestion
    Block & block = probe_process_info.block;
```

Reply:
my mistake, can not be left reference here.

Reply:
> Block & block = probe_process_info.block;

we can not use reference here, because the probe_process_info.block should not be changed.